```ts
import type { DatePickerModel } from 'v-calendar/dist/types/src/use/datePicker'
```

This will work only in `"moduleResolution": "Node"` the new `Bundler` mode will throw errors

```
Cannot find module 'v-calendar/dist/types/src/use/datePicker' or its corresponding type declarations.
```

---

Can you extract the type from `v-calendar` and put the type inside the `Calendar.vue`?

---

Looks like the main `shadcn-ui` is going to add the new Calendar implementation with `@internationalized/date` we can inspire from them once it's ready

--- a/apps/www/src/lib/registry/default/ui/calendar/Calendar.vue
+++ b/apps/www/src/lib/registry/default/ui/calendar/Calendar.vue
@@ -4,9 +4,26 @@ import type { Calendar } from 'v-calendar'
 import { DatePicker } from 'v-calendar'
 import { ChevronLeft, ChevronRight } from 'lucide-vue-next'
 import { computed, nextTick, onMounted, ref } from 'vue'
-import type { DatePickerModel } from 'v-calendar/dist/types/src/use/datePicker'
 import { buttonVariants } from '@/lib/registry/default/ui/button'
 import { cn } from '@/lib/utils'
+
+/* Extracted from v-calendar */
+type DatePickerModel = DatePickerDate | DatePickerRangeObject
+type DateSource = Date | string | number
+type DatePickerDate = DateSource | Partial<SimpleDateParts> | null
+interface DatePickerRangeObject {
+  start: Exclude<DatePickerDate, null>
+  end: Exclude<DatePickerDate, null>
+}
+interface SimpleDateParts {
+  year: number
+  month: number
+  day: number
+  hours: number
+  minutes: number
+  seconds: number
+  milliseconds: number
+}
 
 defineOptions({
   inheritAttrs: false,

--- a/apps/www/src/lib/registry/new-york/ui/calendar/Calendar.vue
+++ b/apps/www/src/lib/registry/new-york/ui/calendar/Calendar.vue
@@ -4,14 +4,30 @@ import type { Calendar } from 'v-calendar'
 import { DatePicker } from 'v-calendar'
 import { ChevronLeftIcon, ChevronRightIcon } from '@radix-icons/vue'
 import { computed, nextTick, onMounted, ref } from 'vue'
-import type { DatePickerModel } from 'v-calendar/dist/types/src/use/datePicker'
 import { buttonVariants } from '@/lib/registry/new-york/ui/button'
 import { cn } from '@/lib/utils'
+
+/* Extracted from v-calendar */
+type DatePickerModel = DatePickerDate | DatePickerRangeObject
+type DateSource = Date | string | number
+type DatePickerDate = DateSource | Partial<SimpleDateParts> | null
+interface DatePickerRangeObject {
+  start: Exclude<DatePickerDate, null>
+  end: Exclude<DatePickerDate, null>
+}
+interface SimpleDateParts {
+  year: number
+  month: number
+  day: number
+  hours: number
+  minutes: number
+  seconds: number
+  milliseconds: number
+}
 
 defineOptions({
   inheritAttrs: false,
 })
-
 const props = withDefaults(defineProps< {
   modelValue?: string | number | Date | DatePickerModel
   modelModifiers?: object

